modules/oai.xql

```diff
@@ -0,0 +1,136 @@
+xquery version "3.1" encoding "UTF-8";
+
+(:~
+ :  Module for exporting data as oai/xml
+ :  see https://www.openarchives.org/OAI/openarchivesprotocol.html
+ :)
+
+declare namespace tei="http://www.tei-c.org/ns/1.0";
+declare namespace mei="http://www.music-encoding.org/ns/mei";
+declare namespace output="http://www.w3.org/2010/xslt-xquery-serialization";
+declare namespace request="http://exist-db.org/xquery/request";
+declare namespace response="http://exist-db.org/xquery/response";
+declare namespace util="http://exist-db.org/xquery/util";
+declare namespace oai="http://www.openarchives.org/OAI/2.0/";
+
+import module namespace functx="http://www.functx.com";
+import module namespace crud="http://xquery.weber-gesamtausgabe.de/modules/crud" at "crud.xqm";
+import module namespace config="http://xquery.weber-gesamtausgabe.de/modules/config" at "config.xqm";
+import module namespace wega-util="http://xquery.weber-gesamtausgabe.de/modules/wega-util" at "wega-util.xqm";
+import module namespace date="http://xquery.weber-gesamtausgabe.de/modules/date" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/date.xqm";
+import module namespace str="http://xquery.weber-gesamtausgabe.de/modules/str" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/str.xqm";
+import module namespace mycache="http://xquery.weber-gesamtausgabe.de/modules/cache" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/cache.xqm";
+import module namespace lod="http://xquery.weber-gesamtausgabe.de/modules/lod" at "lod.xqm";
+import module namespace lang="http://xquery.weber-gesamtausgabe.de/modules/lang" at "lang.xqm";
+import module namespace query="http://xquery.weber-gesamtausgabe.de/modules/query" at "query.xqm";
+import module namespace wdt="http://xquery.weber-gesamtausgabe.de/modules/wdt" at "wdt.xqm";
+
+declare option output:method "xml";
+declare option output:media-type "application/xml";
+declare option output:indent "yes";
+
+(:~
+ : Get the last date of modification from dataHistory.xml. Fallback: VersionDate from options.xml
+ :
+ : @author Dennis Ried 
+:)
+declare variable $oai:last-modified as xs:dateTime? := 
```

Comment:
I see several issues here: 
1. this does not retrieve the last-modified date for the individual record but for the whole data set?
2. better use `config:get-svn-props($docID as xs:string)` which will provide this information for a given docID. 
3. I think we agreed that we do not need the else-case. Better to provide a proper dataHistory file?

Aside: you probably do not need this tokenize-functxdateTime-dance. Just use `config:get-option('versionDate') => xs:dateTime()` (and check `castable as` before)

Reply:
The else case is now necessary, because the response-header-date-function must not get an empty value. So if no modification date is found the current date will be retuned to avoid failing.

I fixed the other issues, hopefully.